### DISCUSSION TITLE
geometric multigrid

### DISCUSSION THREAD

**Discussion by jrobcary** [trilinos/Trilinos#12780] [Solvers]
Is there a geometric multigrid solver in Trilinos?  I cannot find one, and my highly varying grid Poisson problem is defeating ML.

**Comment by GrahamBenHarper** [comment]
Hi @jrobcary, do you mean geometric multigrid in the traditional sense with rediscretized operators on each level? We have some coordinates-informed capabilities in MueLu that might be helpful, but not traditional geometric multigrid.

**Reply by jrobcary** [reply]
@GrahamBenHarper, I don't know, for certain.  But here is a picture of the mesh.  It's a Cartesian product, but both axis are highly refined in the center, such that cells in the center are square but small, cells at the corners are square and large, the other cells can be highly oblong.  The desired mesh has a variation of 10^4 in cell lengths.

The AMG solvers I have tried so far, including ML, take many iterations to converge.

I know the solution has small structure at the center and large as one moves out, so I believe a geometric multigrid-like smoothing/reduction would work, but I cannot figure out how to tell ML or MueLu to do this.  Vanilla settings do not work well.

And yes, I mean the averaging of nodes that geometric multigrid would end up doing, not looking at any coefficients.  Just 4 cells to 1 stuff, then prolongation.

![Bx](https://github.com/trilinos/Trilinos/assets/1441100/72d446f0-f235-4561-bf7e-abcf543c693b)

**Comment by jrobcary** [comment]
I found https://bitbucket.org/hpgmg/hpgmg, but the last commit was 5 years ago.
